python/ray/autoscaler/node_provider.py

```diff
@@ -226,7 +232,14 @@ def get_command_runner(self, log_prefix, node_id, auth_config,
             in the CommandRunner. E.g., subprocess.
         use_internal_ip(bool): whether the node_id belongs to an internal ip
             or external ip.
+        docker_config(dict): If set, the docker information of the docker
+            container that commands should be run on.
         """
-
-        return SSHCommandRunner(log_prefix, node_id, self, auth_config,
-                                cluster_name, process_runner, use_internal_ip)
+        if docker_config and docker_config["container_name"] != "":
+            return DockerCommandRunner(docker_config, log_prefix, node_id,
+                                       self, auth_config, cluster_name,
+                                       process_runner, use_internal_ip)
+        else:
+            return SSHCommandRunner(log_prefix, node_id, self, auth_config,
+                                    cluster_name, process_runner,
+                                    use_internal_ip)
```

Comment:
Can you build the args with a dictionary? 

```
common_args = {...}
if should_use_docker(docker_config):
    return DockerCommandRunner(docker_config=docker_config, **common_args)
...
```

Reply:
Oh--sure. I didn't realize I could do that!